src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -118,7 +113,8 @@ public Task InitializeAsync(ProtectedPrerenderComponentApplicationStore store, C
                     for (var i = 0; i < count; i++)
                     {
                         var (componentType, parameters, sequence) = Descriptors[i];
-                        await Renderer.AddComponentAsync(componentType, parameters, sequence.ToString(CultureInfo.InvariantCulture));
+                        var componentId = Renderer.AddRootComponent(componentType, sequence.ToString(CultureInfo.InvariantCulture));
+                        await Renderer.RenderRootComponentAsync(componentId, parameters);
```

Comment:
In all three of the hosting models, I've done a bit of internal rejiggling to have a consistent "add synchronously, then set initial parameters asynchronously" pattern. It doesn't affect the public API. This is because we need to separate the "add" from "run first render which can take arbitrarily long" parts of the process in order that the dynamic root components APIs make sense.

Since this doesn't affect the existing public API in any way, I don't think there's any drawback to this change, and it's nice that all three models now have identical component-adding concepts.

I did spend ages factoring out the commonality into a whole new layer, `WebRenderer`, but eventually concluded that it didn't give us anything new besides saving a *tiny* amount of similar code, and it forced there to be more public API (due to layering) that I didn't want.